Documentation/design/renderphase.md

```diff
@@ -0,0 +1,133 @@
+# Render
+
+## Goals
+
+1. Render is not required to be run separately. This is exposed as a means of allowing users a deeper entry point for install-time customization.
+
+2. This renders all the assets necessary to bring up a cluster on to disk. This allows users to customize assets without over-parameterizing the `InstallConfig`.
+
+3. Allow user to commit the assets that describe a cluster.
+
+## Non-goals
+
+1. Render is incapable of allowing users to customize cluster objects beyond those already supported by the object's operator.
+
+## Overview
+
+Render has 3 steps:
+
+1. Validate `InstallConfig` object. If `install-config.yaml` file is not present fallback to init phase to generate `InstallConfig` object.
+
+2. Generate all the required TLS assets for the cluster.
+
+3. Use `operator render self` interface to generate the cluster objects for all the operators.
```

Comment:
As a TODO we should probably describe this in more detail / reasoning for operators implementing "render self"  (can be link to another doc if needed)

Reply:
https://github.com/openshift/installer/pull/49/files#diff-43314b7946739da620d1f68378486627R101

Already have a TODO to choose a preferred solutions and describe it in detail.